### PR TITLE
Quiz view

### PR DESCRIPTION
**Pull request by JannatPapia**
Let's  add a contributing file, so we can better work together!

### CODE REVIEW

File: QuizAppNew/Model/QuestionModel.swift [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+//  QuestionModel.swift
+//  QuizAppNew
+//
+//  Created by Jannatun Nahar Papia on 20/8/22.
+//
+
+import SwiftUI
+
+//MARK: Codeble Model....
+
```
Comment (nazmulkp) [inline QuizAppNew/Model/QuestionModel.swift]: why this model? please remove the model for this project. only focus on UI design with dummy text.

File: QuizAppNew/View/HomeView.swift [hunk 2]
```diff
@@ -8,13 +8,79 @@
 import SwiftUI
 
 struct HomeView: View {
+    
+    @State var show = false
+ // Storing level for fetching questions....
+    @State var set = "Round_1"
+    
+    // for analytices.....
```
Comment (nazmulkp) [inline QuizAppNew/View/HomeView.swift]: Why correct, wrong, answer parameter why need it. You are only responsible for making UI design.

File: QuizAppNew/View/QAview.swift [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+//  QuizAppNew
+//
+//  Created by Jannatun Nahar Papia on 20/8/22.
+//
+
+import SwiftUI
+
+//MARK: Question Answer view
+struct QAview: View {
+    //outgoing
```
Comment (nazmulkp) [inline QuizAppNew/View/QAview.swift]: Why parameters such as correct, wrong, answered, set, data why need it. You are only responsible for making UI design.

File: QuizAppNew/View/QAview.swift [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    @Binding var wrong: Int
+    @Binding var answered: Int
+    var set : String
+    @State var data = Qustion()
+    
+//    @State var answered: Int = 0
+    var body: some View {
+        
+        ZStack {
+
```
Comment (nazmulkp) [inline QuizAppNew/View/QAview.swift:28]: **data.question.isEmpty** why logic in side if remove it. no longer need a progress view.  you are only responsible for making UI design.

File: QuizAppNew/View/QAview.swift [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+            Capsule()
+                .fill(Color.green)
+                .frame(width: 100, height: 6)
+        })
+        .padding([.leading,.trailing,.top],10)
+        
+        //MARK: Correct and Wrong count....
+        HStack{
+            Label(
+               title: { Text(correct == 0 ? "" : "\(correct)")
```
Comment (nazmulkp) [inline QuizAppNew/View/QAview.swift]: why set business logic correct == 0 , you are only responsible for making UI desig

File: QuizAppNew/View/QAview.swift [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+            
+        }
+        .padding([.leading,.trailing,.top],10)
+        
+        
+        
+        ZStack{
+            ForEach("Nazmul hasan".reversed().indices) { index in
+                //View .....
+
```
Comment (nazmulkp) [inline QuizAppNew/View/QAview.swift]: remove death code

File: QuizAppNew/View/QAview.swift [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+                
+    //            QuestionView( correct: $correct, wrong: $wrong, answered: $answered)
+            }
+        }
+        .padding()
+        
+            Spacer(minLength: 0)
+            }
+    }
+        // MARK: fetching......
```
Comment (nazmulkp) [inline QuizAppNew/View/QAview.swift]: why do you call the onAppear method? you are only responsible for making UI design.

File: QuizAppNew/View/QuestionView.swift [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+//  QuestionView.swift
+//  QuizAppNew
+//
+//  Created by Jannatun Nahar Papia on 21/8/22.
+//
+
+import SwiftUI
+
+struct QuestionView: View {
+
```
Comment (nazmulkp) [inline QuizAppNew/View/QuestionView.swift]: Why does the below parameter? use dummy text.

File: QuizAppNew/View/QuestionView.swift [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+                .fontWeight(.heavy)
+                .foregroundColor(Color.black)
+                .padding(.top, 25)
+            
+            Spacer(minLength: 0)
+            
+            // Buttons
+        //    Button(action: {selected = question.optionA}, label: { // right
+            Button(action: {}, label: {
+                Text(question.optionA)
```
Comment (nazmulkp) [inline QuizAppNew/View/QuestionView.swift]: are you stupid? I told you only UI design why use model with property gets option text. use dummy text.

File: QuizAppNew/View/QuestionView.swift [hunk 10]
```diff
@@ -0,0 +1,182 @@
...
+                    Text("Submit")
+                        .fontWeight(.heavy)
+                        .foregroundColor(.white)
+                        .padding(.vertical)
+                        .frame(maxWidth: .infinity)
+                        .background(Color.blue)
+                        .cornerRadius(15)
+                })
+                
+                // disabling
```
Comment (nazmulkp) [inline QuizAppNew/View/QuestionView.swift]: Remove all this logic. your response to making UI design not business logic

File: QuizAppNew/View/QuestionView.swift [hunk 11]
```diff
@@ -0,0 +1,182 @@
...
+            
+        }
+        .padding()
+        .background(Color.white)
+        .cornerRadius(25)
+        .shadow(color: Color.black.opacity(0.05), radius: 5, x: 5, y: 5)
+        .shadow(color: Color.black.opacity(0.05), radius: 5, x: -5, y: -5)
+    }
+    
+    //MARK: highlighting answer use color
```
Comment (nazmulkp) [inline QuizAppNew/View/QuestionView.swift]: are you an idiot why use business logic?

File: QuizAppNew/View/QuestionView.swift [hunk 12]
```diff
@@ -0,0 +1,182 @@
...
+            // displying right if wrong selected
+            if question.isSubmitted && option != selected {
+                if question.answer == option{return Color.green}
+            }
+            return Color.gray
+        }
+    }
+    
+    //MARK: check right or wrong answer
+
```
Comment (nazmulkp) [inline QuizAppNew/View/QuestionView.swift]: why logic?

File: QuizAppNew/Model/QuestionModel.swift (hunk 1 above)
Comment (nazmulkp) [inline QuizAppNew/Model/QuestionModel.swift]: Why this model? Why need it? please remove this model. you are responsible for making UI design.

**Review by nazmulkp — CHANGES_REQUESTED** [review]

File: QuizAppNew/Model/QuestionModel.swift (hunk 1 above)
Comment (JannatPapia) [reply QuizAppNew/Model/QuestionModel.swift]: remove model

File: QuizAppNew/Model/QuestionModel.swift [hunk 13]
```diff
@@ -7,31 +7,31 @@
 
 import SwiftUI
```
Comment (nazmulkp) [inline QuizAppNew/Model/QuestionModel.swift]: Deleted this file. don't commend.

**Review by nazmulkp — CHANGES_REQUESTED** [review]

File: QuizAppNew/Model/QuestionModel.swift (hunk 1 above)
Comment (nazmulkp) [reply QuizAppNew/Model/QuestionModel.swift]: deleted the file.

File: QuizAppNew/View/HomeView.swift (hunk 2 above)
Comment (JannatPapia) [reply QuizAppNew/View/HomeView.swift]: correct wrong selected variable is removed

File: QuizAppNew/View/QAview.swift (hunk 3 above)
Comment (JannatPapia) [reply QuizAppNew/View/QAview.swift]: removed

File: QuizAppNew/View/QAview.swift (hunk 4 above)
Comment (JannatPapia) [reply QuizAppNew/View/QAview.swift:28]: logic is remove .. Only making UI

File: QuizAppNew/View/QuestionView.swift (hunk 9 above)
Comment (JannatPapia) [reply QuizAppNew/View/QuestionView.swift]: UI design is Added

File: QuizAppNew/View/QuestionView.swift (hunk 10 above)
Comment (JannatPapia) [reply QuizAppNew/View/QuestionView.swift]: All business logic is removed

File: QuizAppNew/View/QuestionView.swift (hunk 11 above)
Comment (JannatPapia) [reply QuizAppNew/View/QuestionView.swift]: removed

File: QuizAppNew/View/QuestionView.swift (hunk 12 above)
Comment (JannatPapia) [reply QuizAppNew/View/QuestionView.swift]: Logic is Removed

File: QuizAppNew/Model/QuestionModel.swift (hunk 1 above)
Comment (JannatPapia) [reply QuizAppNew/Model/QuestionModel.swift]: okay I understand

File: QuizAppNew/Model/QuestionModel.swift (hunk 13 above)
Comment (JannatPapia) [reply QuizAppNew/Model/QuestionModel.swift]: okay ..deleting file

**Comment by nazmulkp** [comment]
Remove all **comments in** the code. I mean removing the **death code** from the project makes it cleaner.

**Comment by JannatPapia** [comment]
Removed All comment code .. please check last commit

**Comment by nazmulkp** [comment]
Please fix the **code** alignment. it's hard to read the code.

**Comment by JannatPapia** [comment]
okay .. fixing code alignment

**Comment by JannatPapia** [comment]
Reformatted code easy to read.

File: QuizAppNew/View/HomeView.swift [hunk 14]
```diff
@@ -8,13 +8,59 @@
...
+                        show.toggle()
+                    }
+                }
+            })
+            .padding()
+            Spacer(minLength: 0)
+        }
+        .background(Color.black.opacity(0.05))
+        .ignoresSafeArea()
+        .sheet(isPresented: $show, content: {
```
Comment (nazmulkp) [inline QuizAppNew/View/HomeView.swift]: Don' t use sheet use **fullScreenCover** or **navigation push**

File: QuizAppNew/View/QuestionView.swift [hunk 15]
```diff
@@ -0,0 +1,134 @@
...
+                Button(action: {
+                    self.showLoginView.toggle()
+                }) {
+                    Text("Submit")
+                        .fontWeight(.heavy)
+                        .foregroundColor(.white)
+                        .padding(.vertical)
+                        .frame(maxWidth: .infinity)
+                        .background(Color.blue)
+                        .cornerRadius(15)
```
Comment (nazmulkp) [inline QuizAppNew/View/QuestionView.swift:85]: Don't use **sheet** use if-else logic to show the result view.

**Review by nazmulkp — CHANGES_REQUESTED** [review]

File: QuizAppNew/View/QAview.swift [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+//
+//  Created by Jannatun Nahar Papia on 20/8/22.
+//
+
+import SwiftUI
+
+//MARK: Question Answer view
+struct QAview: View {
+    @Environment(\.presentationMode) var present
+    var body: some View {
```
Comment (nazmulkp) [inline QuizAppNew/View/QAview.swift:13]: use if else in QAView instead **QuestionView**
```
if isSubmited {
       ResultView()
}else{
    //rest of UI question  view code. 
}
```

**Review by nazmulkp — CHANGES_REQUESTED** [review]

File: QuizAppNew/View/QAview.swift [hunk 17]
```diff
@@ -9,8 +9,12 @@ import SwiftUI
 
 //MARK: Question Answer view
 struct QAview: View {
+    @State var isSubmitted = false
     @Environment(\.presentationMode) var present
     var body: some View {
+        if isSubmitted {
```
Comment (nazmulkp) [inline QuizAppNew/View/QAview.swift:15]: You can't use **if else** direct in **var body: some View {** without SwiftUI parent Building layouts such as HStack, VStack, ZStack etch

**Review by nazmulkp — CHANGES_REQUESTED** [review]

File: QuizAppNew/View/QuestionView.swift [hunk 18]
```diff
@@ -0,0 +1,142 @@
...
+                
+                //                Button("Submit") {
+                //                          self.showLoginView = true
+                //                      }
+                //                .sheet(isPresented: $showLoginView, content: {
+                //                          ResultView()
+                //                      })
+                //           //     Text("Submit")
+                Button(action: {
+                    self.destination = QAview()
```
Comment (nazmulkp) [inline QuizAppNew/View/QuestionView.swift]: **self.destination = QAview()**   what is this? how does it work?. Writing something will work?

You know how to hide instagram and how to act.

**Review by nazmulkp — CHANGES_REQUESTED** [review]

File: QuizAppNew/View/QAview.swift [hunk 19]
```diff
@@ -0,0 +1,67 @@
...
+//
+//  Created by Jannatun Nahar Papia on 20/8/22.
+//
+
+import SwiftUI
+
+//MARK: Question Answer view
+struct QAview: View {
+    @State var isSubmitted = false
+    @Environment(\.presentationMode) var present
```
Comment (nazmulkp) [inline QuizAppNew/View/QAview.swift]: **@Environment(\.presentationMode) var present** why need it here?

**Review by nazmulkp — CHANGES_REQUESTED** [review]

File: QuizAppNew/View/QuestionView.swift [hunk 20]
```diff
@@ -0,0 +1,141 @@
...
+                //
+                //                                Button("Submit") {
+                //                                    view.isSubmitted  = true
+                //                                      }
+                //                .sheet(isPresented: $showLoginView, content: {
+                //                          ResultView()
+                //                      })
+                //           //     Text("Submit")
+                Button(action: {
+                    showLoginView.toggle()
```
Comment (nazmulkp) [inline QuizAppNew/View/QuestionView.swift]: **showLoginView** is there any login functionality? Please gives the meaningful variable name.

File: QuizAppNew/View/QuestionView.swift [hunk 21]
```diff
@@ -0,0 +1,141 @@
...
+                    .background(
+                        RoundedRectangle(cornerRadius: 15)
+                            .stroke(Color.black.opacity(0.5), lineWidth: 1)
+                    )
+            })
+            
+            Spacer(minLength: 0)
+            
+            HStack(spacing: 15) {
+                /* ---------
```
Comment (nazmulkp) [inline QuizAppNew/View/QuestionView.swift]: remove the death code.

File: QuizAppNew/View/ResultView.swift [hunk 22]
```diff
@@ -0,0 +1,64 @@
...
+//  ResultView.swift
+//  QuizAppNew
+//
+//  Created by Jannatun Nahar Papia on 21/8/22.
+//
+
+import SwiftUI
+
+struct ResultView: View {
+        @Environment(\.presentationMode) var present
```
Comment (nazmulkp) [inline QuizAppNew/View/ResultView.swift]: Fix the alignment.

**Review by nazmulkp — CHANGES_REQUESTED** [review]

**Review by nazmulkp — APPROVED** [review]

**Pull request merged**